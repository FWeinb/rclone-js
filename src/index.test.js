import { Rclone } from './index';

test('provide access to file and path ciphers', done => {
  Rclone({
    password: '',
    salt: ''
  }).then(rclone => {
    expect(rclone).toMatchSnapshot();
    done();
  });
});

test('should return keys using getKeys()', done => {
  Rclone({ password: '', salt: '' }).then(rclone => {
    const keys = rclone.getKeys();
    expect(keys.dataKey).toBeInstanceOf(Uint8Array);
    expect(keys.nameKey).toBeInstanceOf(Uint8Array);
    expect(keys.nameTweak).toBeInstanceOf(Uint8Array);
    expect(keys.password).toEqual('');
    expect(keys.salt).toEqual('');
    done();
  });
});
