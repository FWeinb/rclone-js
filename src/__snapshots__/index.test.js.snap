// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`provide access to file and path ciphers 1`] = `
Object {
  "File": Object {
    "calculateDecryptedSize": [Function],
    "createReadStream": [Function],
    "createReadStreamFactory": [Function],
  },
  "Path": Object {
    "decrypt": [Function],
    "decryptName": [Function],
    "encrypt": [Function],
    "encryptName": [Function],
  },
  "getKeys": [Function],
}
`;
